currency-rates/internal/services/subscription.go

```diff
@@ -55,7 +55,7 @@ func (service SubscriptionService) NotifySubscribers(ctx context.Context) error
 		actualEmails = append(actualEmails, email.Email)
 	}
 
-	err = service.notificationClient.Send(ctx, rate, actualEmails)
+	err = service.notification.Publish(ctx, rate, actualEmails)
```

Comment:
So here you should form your message for publisher, not inside publisher itself.